packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -406,6 +405,36 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
     `, [userId]);
   }
 
+  async getDialogueRecommendedUsers(userId: string, upvotedUsers:UpvotedUser[], limit = 100): Promise<DbUser[]> {
+  const upvotedUserIds = upvotedUsers.map(user => user._id);
+
+  return this.any(`
+    SELECT u.*
+    FROM unnest($2::text[]) AS uv(_id)
+    INNER JOIN "Users" AS u ON uv._id = u._id
+    WHERE
+      -- Don't recommend you users who've never commented twice your posts OR replied 3 times to your comments 
+      EXISTS (
+        SELECT 2
+        FROM public."Comments" AS c
+        INNER JOIN "Posts" AS p ON c."postId" = p._id
+        WHERE c."userId" = uv._id AND p."userId" = $1
+      )
+      OR (
+        SELECT COUNT(*) >= 3
+        FROM public."Comments"
+        WHERE
+          "userId" = uv._id
+          AND "parentCommentId" IN (
+            SELECT _id
+            FROM "Comments"
+            WHERE "userId" = $1
+          )
+      )
+    LIMIT $3 
+  `, [userId, upvotedUserIds, limit]);
+  }
```

Comment:
I think I remember this getting a comment from Ollie on the previous PR... would go back and check